### PR TITLE
FEATURE: allows to force a thread

### PR DESCRIPTION
**Pull request by jjaffeux**
Forcing a thread will work even in channel which don't have `threading_enabled` or in direct message channels.

For now this feature is only available through the `ChatSDK`:

```ruby
ChatSDK::Message.create(
  in_reply_to_id: 1, 
  guardian: guardian, 
  raw: "foo bar baz", 
  channel_id: 2, 
  force_thread: true
)
```

One notable limit of this for now is that we won't show the "My Threads" link in the sidebar, so the forced thread will only be available from the thread preview in the channel.

### CODE REVIEW

File: plugins/chat/app/services/chat/create_message.rb [hunk 1]
```diff
@@ -144,7 +146,7 @@ def instantiate_message(channel:, guardian:, contract:, uploads:, thread:, reply
       )
     end
 
-    def save_message(message_instance:, **)
+    def save_message(message_instance:, thread:, **)
```
Comment (dbattersby) [inline plugins/chat/app/services/chat/create_message.rb:149]: I think we can remove `thread:` here as it's not used.

File: plugins/chat/assets/javascripts/discourse/models/chat-thread.js [hunk 2]
```diff
@@ -31,6 +31,7 @@ export default class ChatThread {
   @tracked tracking;
   @tracked currentUserMembership;
   @tracked preview;
+  @tracked force;
```
Comment (dbattersby) [inline plugins/chat/assets/javascripts/discourse/models/chat-thread.js:34]: Does this need to be a tracked property?

File: plugins/chat/spec/services/chat/list_channel_messages_spec.rb [hunk 3]
```diff
@@ -159,10 +159,29 @@
     context "when threads are disabled" do
       fab!(:thread_1) { Fabricate(:chat_thread, channel: channel) }
 
-      before { channel.update!(threading_enabled: false) }
+      before do
+        channel.update!(threading_enabled: false)
+        thread_1.add(user)
+      end
 
       it "returns empty tracking" do
```
Comment (dbattersby) [inline plugins/chat/spec/services/chat/list_channel_messages_spec.rb:167]: I think we can rename this test as it returns the thread tracking state now.

**Review by dbattersby — APPROVED** [review]
Looks good, I added a couple of very small suggestions but otherwise good to go 👍

File: plugins/chat/assets/javascripts/discourse/models/chat-thread.js (hunk 2 above)
Comment (jjaffeux) [reply plugins/chat/assets/javascripts/discourse/models/chat-thread.js:35]: You are right that it's probably not necessary ATM, but for now I think I prefer to consider every public model properties as tracked. It's so unexpected to have something untracked when working with a model

**Pull request merged**